plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -49,10 +49,13 @@
     private Map<String, String> secrets = new HashMap<>();
     private Vault vault;
     private VaultConfig vaultConfig;
-    private VaultAuthenticator vaultAuthenticator;
+    private VaultAuthenticator currentlyUsedAuthenticator;
+    private VaultAuthenticator configuredAuthenticator;
```

Comment:
does this need to be a field? looks like its only used inside the configureVault method,
I think you could inline it and undo the field rename above?

Reply:
You are right, but I wasn't sure how to do it. I am using lambda expressions https://github.com/jenkinsci/configuration-as-code-plugin/pull/810/files#diff-f7f1df35659ecbdcd0652b07007b223fR111 

I am getting the following if I declare it in a local function context: 
```
Variable used in lambda expression should be final or effectively final
```

I could remove the lambda functions, but that would be replaced by lots of `!= null` checks, so I thought declaring `configuredAuthenticator` in class context would be easy way out.